kep/9-capacity-scheduling-for-ml-dl-workloads-based-on-scheduler-framework/README.md

```diff
@@ -0,0 +1,208 @@
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+There is increasing demand to use Kubernetes to manage batch workloads (ML/DL). In those cases, one challenge is to improve cluster utilization while ensuring that each user has a reasonable amount of resources. The problem can be partially addressed by the Kubernetes [ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/). The native Kubernetes ResourceQuota API can be used to specify the maximum overall resource allocation per namespace. The quota enforcement is done through an admission check. A quota resource consumer (e.g., a Pod) cannot be created if the aggregated resource allocation exceeds the quota limit. The Kubernetes quota design has the limitation: the quota resource usage is aggregated based on the resource configurations (e.g., Pod cpu/mem requests specified in the Pod spec). Although this mechanism can guarantee that the actual resource consumption will never exceed the ResourceQuota limit, it might lead to low resource utilization. For instance, actual resource 
+consumption may be much smaller than the limit.
+
+Due to above limitation, the batch workloads (ML/DL) can't run in a Kubernetes cluster as efficiently as they do in other container orchestration platforms such as Yarn. In order to overcome above limitations, we introduce an "ElasticQuota" concept used in [Yarn capacity scheduler](https://www.netjstech.com/2018/04/capacity-scheduler-in-yarn-hadoop.html) in Kubernetes. Basically, the "ElasticQuota" has the notions of "max" and "min", where the "min” is the minimum resources that are needed to ensure the basic functionality/performance of the consumers and the "max" specifies the upper bound of the resource consumption of the consumers. By introducing "min" and "max", capacity scheduling can be optimized as follows:
+
+1. The resource buffer between "min" and "max" can help to tolerate runtime failures. For example, if a Pod that consumes the entire "min" fails to run, new Pods can still be created if the "max" has not been reached. When using Kubernetes resource quota, once the Pod that consumes the entire quota is created,
+no other Pods can be created - even if the Pod failed to be scheduled, or failed on running (error
+when pulling image, etc.).
```

Comment:
_"once the Pod that consumes the entire quota is created,
 no other Pods can be created - even if the Pod failed to be scheduled, or failed on running"_

This sentence is not easy to ready. Do you mean " Once a namespace reaches its quota limit, no Pods can be created, even if there are pods in the namespace that failed to be scheduled or run."?

Reply:
Done. I change it to `once the Pod that consumes the entire quota is created and the namespace reaches its quota limit, no other Pods can be created - even if the Pod failed to be scheduled, or failed on running (error when pulling image, etc.).`